iModelCore/ecobjects/src/ECInstance.cpp

```diff
@@ -2625,7 +2631,11 @@ struct  InstanceXmlReader
                 return nullptr;
                 }
 
-            ECSchemaCP constraintSchema = Utf8String::IsNullOrEmpty(constraintSchemaName.c_str()) ? defaultSchema : GetSchema(constraintSchemaName);
+            ECSchemaCP constraintSchema = Utf8String::IsNullOrEmpty(constraintSchemaName.c_str()) ? defaultSchema : nullptr;

```

Comment:
While we're here looking at this what happens if a relationship constraint is in a pruned schema.  I don't think we will hit this case with the standard set of schemas on the prune list but it's worth checking if we've got a test for that

Reply:
It is improbable to hit this code if the schema is pruned because the code that finds the attributes in the relationship class is further down. We have a test called "SchemaDeserializationTest.PruneSchemas" that checks if the relationship works even when it uses constraints from pruned schema.